src/api/app/models/event_find_subscriptions.rb

```diff
@@ -31,86 +31,76 @@ def subscriptions
   def expand_toconsider
     new_toconsider = []
     @toconsider.each do |subscription|
-      new_toconsider.concat(expand_one_rule(subscription))
+      new_toconsider.concat(expand_subscription(subscription))
     end
     @toconsider = new_toconsider
   end
 
-  def expand_one_rule(r)
-    if r.receiver_role == :all
-      return [r]
-    end
+  # Expand the EventSubscriptions receiver_role.
+  # Receivers can be many Users and Groups. We need to instantiate EventSubscriptions
+  # for all of them.
+  def expand_subscription(subscription_to_expand)
+    # Fetch all User/Groups from the Event that match the receiver_role of
+    # the EventSubscription
+    receivers = @event.send("#{subscription_to_expand.receiver_role}s")
 
-    receivers = @event.send("#{r.receiver_role}s")
+    # Fetch User ids
+    user_ids = receivers.select { |reciver| reciver.kind_of? User }.map(&:id)
 
-    # fetch database settings
-    user_ids = receivers.select { |rcv| rcv.kind_of? User }.map(&:id)
-    groups = receivers.select { |rcv| rcv.kind_of? Group }
+    # Fetch Group ids
+    groups = receivers.select { |reciver| reciver.kind_of? Group }
     group_ids = []
     groups.each do |group|
+      # If the group has an email set we'll consider that one
       if group.email
         group_ids << group.id
       else
-        # it has not, so write to all users individually
-        group.users.each do |u|
-          next unless user_subscribed_to_group_email?(group, u)
-          user_ids << u.id
+        # if the Group has no email set we consider all users of it individually
+        group.users.each do |user|
+          # of course only when the User is "subscribed" to the group
+          next unless user_subscribed_to_group_email?(group, user)
+          user_ids << user.id
         end
       end
     end
 
     table = EventSubscription.arel_table
 
-    rel = EventSubscription.where(eventtype: r.eventtype, receiver_role: r.receiver_role)
-    ret = rel.where(table[:user_id].in(user_ids).or(table[:group_id].in(group_ids))).to_a
-
-    receivers.each do |ug|
-      # add a default
-      nes = EventSubscription.new(eventtype: r.eventtype, receiver_role: r.receiver_role, channel: r.channel)
-      if ug.kind_of? User
-        nes.user = ug
+    # First find all the subscriptions that are in the database for Users/Groups
+    rel = EventSubscription.where(eventtype: subscription_to_expand.eventtype, receiver_role: subscription_to_expand.receiver_role)
+    subscriptions = rel.where(table[:user_id].in(user_ids).or(table[:group_id].in(group_ids))).to_a
```

Comment:
[Nitpick] I would get rid of the `rel` assignment and just chain all these methods together like:
```ruby
subscriptions = 
  EventSubscription.where(eventtype: subscription_to_expand.eventtype, receiver_role: subscription_to_expand.receiver_role)
                   .where(table[:user_id].in(user_ids).or(table[:group_id].in(group_ids))).to_a
```

Reply:
[Nitpick] Also I dont think you need to directly use Arel for this query. Your PR is still a big improvement over what this was like before but if you want to go one step further you can get rid of the arel table and just use plain active record since for the where..in..or clause.

Reply:
I just want everyone to understand this code so we can refactor it completely at some later point. I really don't want to put 💄 on this 🐷...